Utils/github_workflow_scripts/send_slack_message.py

```diff
@@ -0,0 +1,219 @@
+from typing import List
+
+from slack_sdk import WebClient
+from blessings import Terminal
+from utils import get_env_var
+import json
+import requests
+from github import Github, PaginatedList, File, PullRequest
+import urllib3
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+GREEN_COLOR = "#6eb788"
+SLACK_CHANNEL_TO_SEND_PR_TO = 'contribution-reviews'
+
+
+def get_metadata_file(file: File) -> dict:
+    """Perform a GET request to receive a given file content
+
+    Args:
+        file (File): The file object to receive the content for
+
+    Returns:
+        (dict): Content of a metadata file
+    """
+    raw_url = file.raw_url
+    try:
+        response_json = requests.get(raw_url, verify=False).json()
+    except ValueError:
+        raise Exception(f'{file.filename} is not a well-formatted metadata.json file')
+    return response_json
+
+
+def create_slack_markdown(text: str) -> dict:
+    """Create slack block-kit markdown entry
+
+    Args:
+        text (str): String to appear in the entry
+
+    Returns:
+        (dict): markdown entry for the slack block-kit
+    """
+    return {
+        "type": "mrkdwn",
+        "text": text
+    }
+
+
+def create_slack_fields(text_fields: list) -> dict:
+    """Create slack block-kit section entry with fields key
+
+        Args:
+            text_fields (list): String to appear in the entry
+
+        Returns:
+            (dict): section entry for the slack block-kit
+    """
+    return {
+        "type": "section",
+        "fields": [create_slack_markdown(f'*{key}*:\n {value}') for key, value in text_fields]
+    }
+
+
+def create_slack_section(key: str, value: str) -> dict:
+    """Create slack block-kit section entry with text key
+
+        Args:
+            key (str): pack related key (example pack version)
+            value (str): pack related value (example 1.0.0)
+
+        Returns:
+            (dict): section entry for the slack block-kit
+    """
+    return ({
+        "type": "section",
+        "text": create_slack_markdown(f'```{key}: {value}\n```')
+    })
```

Comment:
You can collapse these into one function that accepts a `type` argument of either `fields` or `text`, and the value to assign it.

Reply:
It is possible, tho, in my opinion, this change will only save a few lines of code but will make this part of code a bit less clear.